### PR TITLE
Disable window scale factor buttons when not applicable

### PR DESCRIPTION
**Pull request by seifer7**
Implemented a check when the Plus or Minus button is pressed on the Options window to change the Scale Factor to make sure that the new scale factor is actually different to the current one.
If the new scale factor has not changed (ie. it's outside the expected range) we cancel the function instead of refreshing the Screen/UI and causing a purple flash.

### CODE REVIEW

File: src/openloco/windows/options.cpp [hunk 1]
```diff
@@ -456,7 +456,10 @@ namespace openloco::ui::options
         static void display_scale_mouse_down(window* w, widget_index wi, float adjust_by)
         {
             auto& config = config::get_new();
+            float old_config_scale_factor = config.scale_factor;
             config.scale_factor = std::clamp(config.scale_factor + adjust_by, 1.0f, 4.0f);
+            if (old_config_scale_factor == config.scale_factor)
+                return;
```
Comment (AaronVanGeffen) [inline src/openloco/windows/options.cpp:463]:
```suggestion
            float newScaleFactor = std::clamp(config.scale_factor + adjust_by, 1.0f, 4.0f);
            if (config.scale_factor == newScaleFactor)
                return;

            config.scale_factor = newScaleFactor;
```
```

**Comment by AaronVanGeffen** [comment]
Initially, I was a little confused by this PR's title, as I don't see a way for it to get outside of the expected range due to `std::clamp` being used. However, your patch does address the problem you describe above. That is, if you decrement the window scale while it's already at the minimum value, the canvas is still unnecessarily invalidated.

I would prefer to resolve this in a slightly different manner, though. I have left a suggestion.

**Comment by seifer7** [comment]
Don't merge this yet.
Working on a much better solution, also taking into account your suggestion.

**Comment by seifer7** [comment]
@AaronVanGeffen Okay, done. I think this is a useful way of implementing this.
Please give me any feedback you might have.

The upper and lower limit, plus the "step" size are now defined in:
`openloco::ui::ScaleFactor::min`, `openloco::ui::ScaleFactor::max`, and `openloco::ui::ScaleFactor::step`

On the Options window, the up and down arrows for "Window Scale" adjustment now disable once the upper or lower limit is reached.

I moved the window scale adjustment code out of the `openloco::ui::options::display_scale_mouse_down` button click event, and put it into it's own function called: `openloco::ui::adjust_window_scale`

Now it can be called from anywhere, not just that button. For example for an in-game hotkey to adjust the scale up or down it'd just call `openloco::ui::adjust_window_scale(float adjust_by)`
I tested by replacing the Rotate view hotkey with the above code and if successfully increases the window scale.

**Comment by AaronVanGeffen** [comment]
This conflicts with the changes from #421. Could you rebase this PR? Let us know if you need any help.

**Comment by seifer7** [comment]
@AaronVanGeffen I think I did that right :)

**Comment by AaronVanGeffen** [comment]
@seifer7 You've merged upstream into your branch instead of doing a rebase. However, in this case, that's enough — we can squash-merge this PR. Thanks.

**Pull request merged**